### PR TITLE
fix: remove autodiscovery for otel

### PR DESCRIPTION
**Pull request by glasnt**
## Description

Addresses #74 

In isolated tests, there were improvements to startup performance when introducing this code. 

Before this change: 

```
STARTUP HTTP probe succeeded after 8 attempts for container "server-1" on path "/ready".
```

After this change: 

```
STARTUP HTTP probe succeeded after 4 attempts for container "server-1" on path "/ready".
```

This is based on the log-based output from the startup probe, with a period of 1, this relates to an improvement of "halving" the startup time. 

*results observed in isolation

### CODE REVIEW

**Comment by ymotongpoo** [comment]
Though `opentelemetry-instrument` is handy when you need instrumentation in the app, it requires static code analysis and code injection for instrumentation on launch time. If you have specific end point in mind to instrument, explicitly instrumenting your code is much safer and reasonable when the start up time matters and the number of endpoint won't change much.

File: server/manage.py [hunk 1]
```diff
@@ -19,10 +19,14 @@
 import os
 import sys
 
+from opentelemetry.instrumentation.django import DjangoInstrumentor
 
 def main():
     """Run administrative tasks."""
     os.environ.setdefault("DJANGO_SETTINGS_MODULE", "avocano_api.settings")
+
+    DjangoInstrumentor().instrument()
```
Comment (grayside) [inline server/manage.py:28]: question: Should this be wrapped in additional controls, for example to not run instrumentation in test?

File: server/Procfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-web: opentelemetry-instrument gunicorn --bind 0.0.0.0:$PORT --workers 1 avocano_api.wsgi:application
+web: gunicorn --bind 0.0.0.0:$PORT --workers 1 avocano_api.wsgi:application
```
Comment (grayside) [inline server/Procfile:1]: In moving from auto-instrumentation to inline, it feels like we need to capture some of the context somewhere. Could we add some documentation about instrumentation approach/decisions to the PR?

Comment (glasnt) [reply server/Procfile:1]: Added as a doc, this repo's version of ADR

File: server/manage.py (hunk 1 above)
Comment (glasnt) [reply server/manage.py:28]: Unlike #291, the server unit tests pass even with this. I'll investigate how this works.

File: docs/admin/README.md [hunk 3]
```diff
@@ -8,6 +8,7 @@ As a infrastructure administrator, I want to:
  * [extend this example application.](extending-example.md)
  * [verify changes to this application.](testing-changes.md)
  * [perform management commands.](management-commands.md)
+ * [see telemetry for my application.](opentelemetry.md)
```
Comment (grayside) [inline docs/admin/README.md:11]: issue: opentelemetry.md missing? I don't this in a separate PR or HEAD.

Comment (glasnt) [reply docs/admin/README.md:11]: I didn't commit the doc and now it's lost. Will recreate

Comment (glasnt) [reply docs/admin/README.md:11]: doc recreated

File: docs/admin/opentelemetry.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# Adding Open Telementry Instrumentation
+
+Avocano uses [`opentelemetry-instrumentation-django`](https://github.com/open-telemetry/opentelemetry-python-contrib/tree/main/instrumentation/opentelemetry-instrumentation-django) to add tracing support, visible in the [Trace Explorer](https://console.cloud.google.com/traces/list) in the Google Cloud Console. 
+
+This is configured by adding environment variables to the [Cloud Run service](/provisioning/terraform/service.tf).
```
Comment (grayside) [inline docs/admin/opentelemetry.md:5]: discussion: Nice, so the same env variables are respected either way.

**Review by grayside — APPROVED** [review]
Thanks for adding the additional documentation.

**Pull request merged**